pandas/io/excel/_odfreader.py

```diff
@@ -200,9 +203,12 @@ def _get_cell_value(self, cell, convert_float: bool) -> Scalar:
             cell_value = cell.attributes.get((OFFICENS, "date-value"))
             return pd.to_datetime(cell_value)
         elif cell_type == "time":
-            stamp = pd.to_datetime(str(cell))
-            # error: Item "str" of "Union[float, str, NaTType]" has no attribute "time"
-            return stamp.time()  # type: ignore[union-attr]
+            stamp: pd.Timestamp | NaTType = pd.to_datetime(str(cell))
+            if not isinstance(stamp, NaTType):
+                return stamp.time()
+            else:
+                self.close()
+                raise ValueError(f"Unrecognized time {str(cell)}")
```

Comment:
If changing code in a typing PR, I have some questions...

does it change behavior?
has typing found a latent bug?
is a release note needed?
do we need some additional tests?

Reply:
Given that it is a time, the other solution is to just do a `cast` to `Timestamp` since we know the result won't be a `NaTType` .  I will revert to that with a comment